tools/configen/tests/test_generate.py

```diff
@@ -100,9 +100,38 @@ def test_generated_code() -> None:
         assert False, f"Mismatch between {expected_file} and generated code"
 
 
+def test_generated_code_with_default_flags() -> None:
```

Comment:
Okay, after seeing the corresponding generated code, I think this is too much.
We don't need to re-test everything from scratch. this strategy would lead to an exponential explosion in the number of tests.

Instead, test the new logic directly.
add a few generated class cases, one with convert, one with recursive and one with both.

Reply:
done